subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleHtmlReport.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality;
+
+import org.gradle.api.reporting.SingleFileReport;
+
+/**
+ * The single file HTML report for Checkstyle.
+ */
+public interface CheckstyleHtmlReport extends SingleFileReport {
```

Comment:
Rather than making two new specific interfaces with basically the same method signatures, please make a more generic interface. This will allow us to use that interface to style other reports generated by other tasks as well.